script.js

```diff
@@ -1,4 +1,272 @@
+  Korean: 2
+});
+const GameState = Object.freeze({
+  EnterName: 0,
+  InGame: 1
+});
+const KoreanGameScore = Object.freeze({
+  PlayerWins: 0,
+  ComputerWins: 1
+});
+
+const programState = {
+  userName: null,
+  koreanGameState: {
+    isFinalRound: false,
+    score: -1
+  },
+  gameMode: GameMode.Normal,
+  gameState: GameState.EnterName,
+  gameScore: {
+    wins: 0,
+    losses: 0,
+    draws: 0
+  }
 };
+
+button.addEventListener("click", main);
+instructionsText.innerHTML = "Enter your name:";
+
+function main() {
+  if (programState.gameState === GameState.EnterName) {
+    programState.userName = document.querySelector("#input-sps").value;
+    programState.gameState = GameState.InGame;
+    document.querySelector("#input-sps").value = "";
+    instructionsText.innerHTML = `Hello, ${programState.userName}!` +
+      "<br>" + "Available modes are <code>reversed</code>, or <code>korean</code>. Refresh the page to reset the game." +
+      "<br>" + "Otherwise, enter scissors, paper or stone (enter random for a random choice):";
+  } else {
+    const rawInput = String(document.querySelector("#input-sps").value);
+
+    switch (rawInput) {
+      case "reversed":
+        {
+          if (programState.gameMode !== GameMode.Reversed) {
+            programState.gameMode = GameMode.Reversed;
+            document.querySelector("#input-sps").value = "";
+            instructionsText.innerHTML = `Reversed mode activated! Enter scissors, paper or stone (enter random for a random choice):`;
+          }
+        }
+        break;
+      case "korean":
+        {
+          if (programState.gameMode !== GameMode.Korean) {
+            programState.gameMode = GameMode.Korean;
+            document.querySelector("#input-sps").value = "";
+            instructionsText.innerHTML = `Korean mode activated! Enter scissors, paper or stone (enter random for a random choice):`;
+          }
+        }
+        break;
+      case "random":
+        {
+          const randomIndex = Math.floor(Math.random() * Object.keys(Choices).length);
+          const randomChoice = Object.values(Choices)[randomIndex];
+          runGame(randomChoice);
+        }
+        break;
+      default:
+        runGame(rawInput);
+        break;
+    }
+  }
+}
+
+function runGame(rawInput) {
+  const randomIndex = Math.floor(Math.random() * Object.keys(Choices).length);
+  const computerChoice = Object.values(Choices)[randomIndex];
+
+  let playerInput = null;
+  let winState = null;
+
+  Object.values(Choices).forEach((value) => {
+    if (rawInput.includes(value)) {
+      playerInput = value;
+    }
+  });
+
+  if (playerInput === null) {
+    displayResults("Please enter a valid input");
+  } else {
+    winState = getWinState(playerInput, computerChoice, programState.gameMode);
+    if (programState.gameMode != GameMode.Korean) {
+      calculateScore(playerInput, computerChoice, winState);
+    }
+  }
+}
+
+function getWinState(playerInput, computerChoice, gameMode) {
```

Comment:
getWinState is a quite long, think it could be split further into multiple functions
1. win/loss of the normal sps game
2. reverse game
3. korean game